### PR TITLE
Allow reloads of Enums.

### LINKED ISSUE #10232: Autoreload Enum Error
I'm getting a weird error when trying to use autoreload and Enums. I have an Enum class defined in a file as:
```python
class MyEnum(Enum):
    A = 'A'
    B = 'B'
```
Using `%autoreload 2`, if I try to add `C = 'C'` to `MyEnum`, or remove an existing definition, I get the following error:
```
[autoreload of updates_type failed: Traceback (most recent call last):
  File "/home/amarder/anaconda3/lib/python3.5/site-packages/IPython/extensions/autoreload.py", line 247, in check
    superreload(m, reload, self.old_objects)
RecursionError: maximum recursion depth exceeded
]
```
Sometimes it is a different error:
```
[autoreload of updates_type failed: Traceback (most recent call last):
  File "/home/amarder/anaconda3/lib/python3.5/site-packages/IPython/extensions/autoreload.py", line 247, in check
    superreload(m, reload, self.old_objects)
AttributeError: __abstractmethods__
]
```

I'm running the code on Debian Linux, using Anaconda with python 3.5.

### PR DESCRIPTION
**Pull request by Carreau**
Closes #10232

### CODE REVIEW

File: IPython/extensions/tests/test_autoreload.py [hunk 1]
```diff
@@ -126,6 +127,31 @@ def new_module(self, code):
...
+            self.write_file(mod_fn, textwrap.dedent("""
+                                    from enum import Enum
+                                    class MyEnum(Enum):
+                                        A = 'A'
+                                        B = 'B'
+                                        C = 'C'
+                                """))
+            with tt.AssertNotPrints(('[autoreload of %s failed:' % mod_name), channel='stderr'):
+                self.shell.run_code("pass")  # trigger another reload
+        except ImportError:
```
Comment (takluyver) [inline IPython/extensions/tests/test_autoreload.py]: Given that autoreload uses import machinery, let's not catch `ImportError` around this test. ;-) We can just catch it on `import enum` at first and skip the test if that fails.

**Comment by Carreau** [comment]
replaced by a  skipif < 3.6

**Comment by Carreau** [comment]
Thanks !

**Comment by Apteryks** [comment]
Hi, and thanks for fixing this problem :). Is it planned for this change to be backported to 5.X version in the future?

**Comment by Carreau** [comment]
> Hi, and thanks for fixing this problem :). Is it planned for this change to be backported to 5.X version in the future?

Backport to 5.x has not beed requested (as we sort of assume user of Python 3 will have 6.x branch). but if you submit a PR against the 5.x branch that backport this fix, we'll be happy to merge it.

**Pull request merged**